clusterdock/topologies/cdh/actions.py

```diff
@@ -189,19 +255,19 @@ def restart_cm_server(primary_node):
     logger.info('Restarting CM server...')
     primary_node.ssh('service cloudera-scm-server restart')
 
-def restart_cm_agents(cluster):
+def restart_cm_agents(cluster, nodes):
```

Comment:
Instead of adding another argument, perhaps just loop through the cluster's nodes and exclude any that we don't want to install on? Same comment applies to functions below that have now had a new `node` argument added. In general, we should always aim to only add args when strictly necessary.